### ISSUE TITLE
Respect XML document encoding

### ISSUE THREAD

**Issue by rgrunber** [redhat-developer/vscode-xml#759]
An XML document can have a document encoding :

`<?xml version="1.0" encoding="UTF-8"?>`

We should detect (entirely on client-side) the type of encoding and set it appropriately. VS Code currently doesn't handle this, so maybe we can support it on our end for now.

**Comment by ds1709** [comment]
I agree. It's very annoying to reopen non UTF-8 xml file every time. Encodimg must be readen from xml file header.

**Comment by angelozerr** [comment]
I fear that vscode doesn't provide an API to update encoding according to this issue https://github.com/microsoft/vscode/issues/824

vscode editorconfig extension is for instance blocked by this API which doesn't exists https://github.com/editorconfig/editorconfig-vscode/issues/35

**Comment by angelozerr** [comment]
@rgrunber @ds1709 as vscode provides now an ecoding API (since 2025), I did a PR to support it https://github.com/redhat-developer/vscode-xml/pull/1166
